pyincore/spaceservice.py

```diff
@@ -73,6 +73,24 @@ def get_space_by_id(self, space_id: str):
 
         return response
 
+    def get_space_by_name(self, space_name: str):
+        """Get space information by querying the name of space.
+
+        Args:
+            space_name (str): A space representation. Name of the Space.
+
+        Returns:
+            obj: HTTP response with the returned space information.
+
+        """
+        r = self.client.get(self.base_space_url, params={"name": space_name})
+        if r.status_code == 200:
+            return r.json()
+        else:
+            # return empty list
+            # todo should this be some error message json instead of empty json?
+            return []
```

Comment:
Looking at other methods, i think they don't check the code and just let the method throw errors

Reply:
changed to throw an error instead of returning an empty list